marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow.sjs

```diff
@@ -140,6 +130,53 @@ class Flow {
     }
   }
 
+  /**
+   * Find records that match a query based on the given inputs. Each matching record is wrapped in a
+   * "content descriptor" object that is guaranteed to have at least a "uri" property.
+   *
+   * @param flowName
+   * @param stepNumber
+   * @param options
+   * @param filterQuery
+   * @return {*}
+   */
+  findMatchingContent(flowName, stepNumber, options, filterQuery) {
+    const flow = this.getFlow(flowName);
+    const flowStep = flow.steps[stepNumber];
+    const stepDetails = this.step.getStepByNameAndType(flowStep.stepDefinitionName, flowStep.stepDefinitionType);
+    const combinedOptions = Object.assign({}, stepDetails.options || {}, flow.options || {}, flowStep.options || {}, options);
+
+    let query;
+    let uris = null;
+    if (options.uris) {
+      uris = this.datahub.hubUtils.normalizeToArray(options.uris);
+      query = cts.documentQuery(uris);
+    } else {
+      let sourceQuery = combinedOptions.sourceQuery || flow.sourceQuery;
+      query = sourceQuery ? xdmp.eval(sourceQuery) : null;
+    }
+
+    if (stepDetails.name === 'default-merging' && stepDetails.type === 'merging' && uris) {
+      return uris.map((uri) => { return { uri }; });
+    } else {
+      let sourceDatabase = combinedOptions.sourceDatabase || this.globalContext.sourceDatabase;
+      if (filterQuery) {
```

Comment:
query can be null at this point? If query and filterQuery are null, what will it happen? I guess we do not want to continue calling queryToContentDescriptorArray. Do we need to add some logs for debugging purpose?